src/Compare/ExtensionConfigReflection.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace SilverStripe\DeprecationChecker\Compare;
+
+use Doctum\Project;
+use Doctum\Reflection\ClassReflection;
+use Doctum\Reflection\PropertyReflection;
+
+class ExtensionConfigReflection extends PropertyReflection implements ExtensionReflection
+{
+    private ?ClassReflection $extensionClass = null;
+
+    public function setExtensionClass(ClassReflection $extensionClass): void
+    {
+        $this->extensionClass = $extensionClass;
+    }
+
+    public function getExtensionClass(): ?ClassReflection
+    {
+        return $this->extensionClass;
+    }
+
+    public static function fromArray(Project $project, array $array): static
+    {
+        // Direct copy/paste from PropertyReflection::fromArray() but swap self for static
+        $property            = new self($array['name'], $array['line']);
```

Comment:
remove all the spaces from before the equal sign, this doesn't match our usual formatting